src/app/components/guards/onboarding/index.tsx

```diff
@@ -23,18 +21,29 @@ function OnboardingGuard({ children }: WalletExistsGuardProps): React.ReactEleme
   useSingleTabGuard('onboarding');
 
   const [walletExistsGuardEnabled, setWalletExistsGuardEnabled] = useState(true);
-
+  const [walletSeedPhrase, setWalletSeedPhrase] = useState<string | null>(null);
+  const { getSeed } = useSecretKey();
   const contextValue: WalletExistsContextProps = useMemo(
     () => ({
       disableWalletExistsGuard: () => setWalletExistsGuardEnabled(false),
     }),
     [],
   );
+  useEffect(() => {
+    (async () => {
+      try {
+        const seed = await getSeed();
+        setWalletSeedPhrase(seed);
```

Comment:
Aren't we unnecessarily putting the seed in memory here? Maybe this should be the `hasSeed` method and we just store a boolean since we aren't actually using the seed?